staging/src/k8s.io/apiserver/pkg/audit/policy/checker_test.go

```diff
@@ -213,3 +214,144 @@ func TestChecker(t *testing.T) {
 	test("subresource", audit.LevelRequest, nil, "getPodLogs", "getPods")
 	test("subresource", audit.LevelRequest, nil, "getPods", "getPodLogs")
 }
+
+func TestCheckerPolicyOmitStages(t *testing.T) {
+	globalStages := []audit.Stage{audit.StageRequestReceived, audit.StageResponseStarted}
+	test := func(req string, expLevel audit.Level, expOmitStages []audit.Stage, ruleNames ...string) {
+		policy := audit.Policy{OmitStages: globalStages}
+		for _, rule := range ruleNames {
+			require.Contains(t, rules, rule)
+			policy.Rules = append(policy.Rules, rules[rule])
+		}
+		require.Contains(t, attrs, req)
+		actualLevel, actualOmitStages := NewChecker(&policy).LevelAndStages(attrs[req])
+		assert.Equal(t, expLevel, actualLevel, "request:%s rules:%s", req, strings.Join(ruleNames, ","))
+		assert.True(t, stageEqual(expOmitStages, actualOmitStages), "request:%s rules:%s, expected stages: %v, actual stages: %v",
+			req, strings.Join(ruleNames, ","), expOmitStages, actualOmitStages)
+	}
+
+	test("namespaced", audit.LevelMetadata, globalStages, "default")
```

Comment:
You can extract the method signature to a named type and then extract this copypasta to a separate method that takes the `test` function

Then you don't even need the separate method, just call this new method twice with a different set of global stages

Reply:
Done.